integration-tests/src/test/java/org/apache/fineract/integrationtests/client/ClientSearchTest.java

```diff
@@ -0,0 +1,178 @@
+        // when
+        PageClientSearchData result = clientHelper.searchClients(request2.getExternalId());
+        // then
+        assertThat(result.getTotalElements()).isEqualTo(1);
+        assertThat(result.getContent().get(0).getExternalId().getValue()).isEqualTo(request2.getExternalId());
+    }
+
+    @Test
+    public void testClientSearchWorks_ByAccountNumber() {
+        // given
+        PostClientsRequest request1 = ClientHelper.defaultClientCreationRequest();
+        clientHelper.createClient(request1);
+
+        PostClientsRequest request2 = ClientHelper.defaultClientCreationRequest();
+        PostClientsResponse response2 = clientHelper.createClient(request2);
+        GetClientsClientIdResponse client2Data = ClientHelper.getClient(requestSpec, responseSpec,
+                Math.toIntExact(response2.getClientId()));
+
+        PostClientsRequest request3 = ClientHelper.defaultClientCreationRequest();
+        clientHelper.createClient(request3);
+        // when
+        PageClientSearchData result = clientHelper.searchClients(client2Data.getAccountNo());
+        // then
+        assertThat(result.getTotalElements()).isEqualTo(1);
+        assertThat(result.getContent().get(0).getAccountNo()).isEqualTo(client2Data.getAccountNo());
+    }
+
+    @Test
+    public void testClientSearchWorks_ByDisplayName() {
+        // given
+        PostClientsRequest request1 = ClientHelper.defaultClientCreationRequest();
+        clientHelper.createClient(request1);
+
+        PostClientsRequest request2 = ClientHelper.defaultClientCreationRequest();
+        clientHelper.createClient(request2);
+        String client2DisplayName = "%s %s".formatted(request2.getFirstname(), request2.getLastname());
+
+        PostClientsRequest request3 = ClientHelper.defaultClientCreationRequest();
+        clientHelper.createClient(request3);
+        // when
+        PageClientSearchData result = clientHelper.searchClients(client2DisplayName);
+        // then
+        assertThat(result.getTotalElements()).isEqualTo(1);
+        assertThat(result.getContent().get(0).getDisplayName()).isEqualTo(client2DisplayName);
+    }
+
+    @Test
+    public void testClientSearchWorks_ByMobileNo() {
+        // given
+        PostClientsRequest request1 = ClientHelper.defaultClientCreationRequest();
+        clientHelper.createClient(request1);
+
+        PostClientsRequest request2 = ClientHelper.defaultClientCreationRequest();
+        request2.setMobileNo(Utils.randomNumberGenerator(8).toString());
+        clientHelper.createClient(request2);
+
+        PostClientsRequest request3 = ClientHelper.defaultClientCreationRequest();
+        clientHelper.createClient(request3);
+        // when
+        PageClientSearchData result = clientHelper.searchClients(request2.getMobileNo());
+        // then
+        assertThat(result.getTotalElements()).isEqualTo(1);
+        assertThat(result.getContent().get(0).getMobileNo()).isEqualTo(request2.getMobileNo());
+    }
```

Comment:
you might wanna add a negative test case, when there is no found item.

Reply:
done